src/opm/output/eclipse/FileService/EclFile.cpp

```diff
@@ -755,6 +758,51 @@ bool EclFile::hasKey(const std::string &name) const
     return search != array_index.end();
 }
 
+std::streampos
+EclFile::seekPosition(const std::vector<std::string>::size_type arrIndex) const
+{
+    if (arrIndex >= this->array_name.size()) {
+        return { static_cast<std::streamoff>(this->ifStreamPos.back()) };
+    }
+
+    // StreamPos is file position of start of data vector's control
+    // character (unformatted) or data items (formatted).  We need
+    // file position of start of header, because that's where we're
+    // supposed to start writing new data.  Subtract header size.
+    //
+    //   (*) formatted header size = 30 characters =
```

Comment:
This is fragile ...

Reply:
> This looks fragile ...

Sure. It's also guaranteed by the protocol.